x509-cert/src/serial_number.rs

```diff
@@ -0,0 +1,74 @@
+//! X.509 serial number
+
+use der::{
+    asn1::Uint, DecodeValue, EncodeValue, ErrorKind, FixedTag, Header, Length, Reader, Result, Tag,
+    ValueOrd, Writer,
+};
+
+const SERIAL_NUMBER_MAX_LEN: Length = Length::new(20);
+
+/// [RFC 5280 Section 4.1.2.2.]  Serial Number
+///
+///   The serial number MUST be a positive integer assigned by the CA to
+///   each certificate.  It MUST be unique for each certificate issued by a
+///   given CA (i.e., the issuer name and serial number identify a unique
+///   certificate).  CAs MUST force the serialNumber to be a non-negative
+///   integer.
+///
+///   Given the uniqueness requirements above, serial numbers can be
+///   expected to contain long integers.  Certificate users MUST be able to
+///   handle serialNumber values up to 20 octets.  Conforming CAs MUST NOT
+///   use serialNumber values longer than 20 octets.
+///
+///   Note: Non-conforming CAs may issue certificates with serial numbers
+///   that are negative or zero.  Certificate users SHOULD be prepared to
+///   gracefully handle such certificates.
+#[derive(Clone, Debug, Eq, PartialEq, ValueOrd, PartialOrd, Ord)]
+pub struct SerialNumber {
+    inner: Uint,
+}
+
+impl SerialNumber {
+    /// Create a new [`SerialNumber`] from a byte slice.
+    pub fn new(bytes: &[u8]) -> Result<Self> {
+        let inner = Uint::new(bytes)?;
+
+        if inner.len() > SERIAL_NUMBER_MAX_LEN {
```

Comment:
I'm introducing this hard limit, which may cause problem trying to parse a certificate not abiding with this limit.
Would love your thoughts on that.

Reply:
Sanity limits seem good to me